direnvrc

```diff
@@ -1,5 +1,122 @@
 # shellcheck shell=bash
 
+_nix_import_env() {
+  local env=$1
+
+  local term_backup=$TERM path_backup=$PATH
+  if [[ -n ${TMPDIR+x} ]]; then
+    local tmp_backup=$TMPDIR
+  fi
+  local impure_ssl_cert_file=${SSL_CERT_FILE:-__UNSET__}
+  local impure_nix_ssl_cert_file=${NIX_SSL_CERT_FILE:-__UNSET__}
+
+  eval "$env"
+
+  # `nix-shell --pure` sets invalid ssl certificate paths
+  if [[ "${SSL_CERT_FILE:-}" = /no-cert-file.crt ]]; then
+    if [[ $impure_ssl_cert_file == __UNSET__ ]]; then
+      unset SSL_CERT_FILE
+    else
+      export SSL_CERT_FILE=$impure_nix_ssl_cert_file
+    fi
+  fi
+
+  if [[ "${NIX_SSL_CERT_FILE:-}" = /no-cert-file.crt ]]; then
+    if [[ $impure_nix_ssl_cert_file == __UNSET__ ]]; then
+      unset NIX_SSL_CERT_FILE
+    else
+      export NIX_SSL_CERT_FILE=${impure_ssl_cert_file}
+    fi
+  fi
+
+  export PATH=$PATH:$path_backup TERM=$term_backup
+  if [[ -n ${tmp_backup+x} ]]; then
+    export TMPDIR=${tmp_backup}
+  else
+    unset TMPDIR
+  fi
+
+  # misleading since we are in an impure shell now
+  export IN_NIX_SHELL=impure
+}
+
+_nix_add_gcroot() {
+  local storepath=$1
+  local symlink=$2
+
+  local stripped_pwd=${PWD/\//}
+  local escaped_pwd=${stripped_pwd//-/--}
+  local escaped_pwd=${escaped_pwd//\//-}
+  ln -fs "$storepath" "$symlink"
+  ln -fs "$symlink" "/nix/var/nix/gcroots/per-user/$USER/$escaped_pwd"
+}
+
+use_flake() {
+  watch_file flake.nix
+  watch_file flake.lock
+
+  local profile="$(direnv_layout_dir)/flake-profile"
+  local profile_rc="${profile}.rc"
+
+  if [[ ! -e "$profile" ]] || \
+    [[ ! -e "$profile_rc" ]] || \
+    [[ "$HOME/.direnvrc" -nt "$profile_rc" ]] || \
+    [[ .envrc -nt "$profile_rc" ]] || \
+    [[ flake.nix -nt "$profile_rc" ]] || \
+    [[ flake.lock -nt "$profile_rc" ]];
+  then
+    local tmp_profile="$(direnv_layout_dir)/flake-profile.$$"
+    [[ -d "$(direnv_layout_dir)" ]] || mkdir "$(direnv_layout_dir)"
+    local tmp_profile_rc=$(nix print-dev-env --profile "$tmp_profile")
+    drv=$(realpath "$tmp_profile")
+    echo "$tmp_profile_rc" > "$profile_rc"
+    rm -f "$tmp_profile" "$tmp_profile"*
+    _nix_add_gcroot "$drv" "$profile"
+    log_status renewed cache
+  else
+    log_status using cached dev shell
+  fi
+
+  local old_nix_build_top=${NIX_BUILD_TOP:-__UNSET__}
+  local old_tmp=${TMP:-__UNSET__}
+  local old_tmpdir=${TMPDIR:-__UNSET__}
+  local old_temp=${TEMP:-__UNSET__}
+  local old_tempdir=${TEMPDIR:-__UNSET__}
+  eval "$(< "$profile_rc")"
+  # nix print-env-dev will create a temporary directory and use it a TMPDIR,
+  # we cannot rely on this directory beeing not deleted at some point,
+  # hence we are just removing it right away.
```

Comment:
I don't follow. We can't a priori rely on `$old_tmpdir` not being deleted either, what's the difference? At least `print-dev-env`'s `TMPDIR` is a subfolder of `/tmp`.

Reply:
This is more about a temporary directory being deleted and we still cache the environment variables pointing to it.

Reply:
`old_tmpdir` might be a value set by a user and hopefully is persistent/sane i.e. TMPDIR=/run/user/$UID or simply TMPDIR=/tmp